matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/notification/DefaultRoomPushRuleService.kt

```diff
@@ -51,7 +51,7 @@ internal class DefaultRoomPushRuleService @AssistedInject constructor(@Assisted
     override fun setRoomNotificationState(roomNotificationState: RoomNotificationState, matrixCallback: MatrixCallback<Unit>): Cancelable {
         return setRoomNotificationStateTask
                 .configureWith(SetRoomNotificationStateTask.Params(roomId, roomNotificationState)) {
-                    this.callback = callback
```

Comment:
This should be treated as an error by the compiler...

Reply:
Very true.

Reply:
Wouldn't happen with suspend functions. :stuck_out_tongue:

Reply:
Why not?

Reply:
With suspend functions the compiler passes the callbacks for you, so you don't get it wrong. (Unless you touch the unsafe low-level utilities)

Reply:
Ah, you mean the `Continuation` object will encapsulate callback. Interesting. Maybe, this would fit here. - Still, I agree with @bmarty that the compiler should warn us a bit harder.

Reply:
Indeed. One day we will probalby use coroutines every where in the SDK API